src/main/java/org/jooq/lambda/SequentialSpliterator.java

```diff
@@ -0,0 +1,39 @@
+package org.jooq.lambda;
+
+import java.util.Spliterator;
+
+/**
+ * Special subinterface of <code>Spliterator</code> that provides default implementations
+ * for all the methods that are not relevant to sequential traversal.
+ *
+ * @see java.util.stream.BaseStream#sequential
+ *
+ * @author Tomasz Linkowski
+ */
+@FunctionalInterface
+public interface SequentialSpliterator<T> extends Spliterator<T> {
```

Comment:
For now, I just placed it inside `TailBuffer`. But maybe it doesn't need to be public, I must say I just added it as public without much consideration.